src/common/lib/lnurl.ts

```diff
@@ -61,27 +67,49 @@ const lnurl = {
 
     return null;
   },
+
   async getDetails(lnurlString: string): Promise<LNURLError | LNURLDetails> {
     const url = normalizeLnurl(lnurlString);
-    let lnurlDetails = {} as LNURLDetails;
-    lnurlDetails.tag = url.searchParams.get("tag") as LNURLDetails["tag"]; // is this still relevant?
-    if (lnurlDetails.tag === "login") {
-      lnurlDetails.k1 = url.searchParams.get("k1") || "";
-      lnurlDetails.action = url.searchParams.get("action") || undefined;
+    const searchParamstag = url.searchParams.get("tag") as LNURLDetails["tag"];
+
+    if (searchParamstag === "login") {
+      const searchParamsK1 = url.searchParams.get(
+        "k1"
+      ) as LNURLAuthServiceResponse["tag"];
+      const searchParamsAction = url.searchParams.get(
+        "action"
+      ) as LNURLAuthServiceResponse["tag"];
+
+      const lnurlAuthDetails: LNURLAuthServiceResponse = {
+        action: searchParamsAction,
+        domain: url.hostname,
+        k1: searchParamsK1,
+        tag: searchParamstag,
+        url: url.toString(),
+      };
+
+      return lnurlAuthDetails;
     } else {
       try {
-        const res = await axios.get(url.toString());
-        lnurlDetails = res.data;
+        const { data }: { data: LNURLDetails | LNURLError } = await axios.get(
+          url.toString()
+        );
+        const lnurlDetails = data;
+
+        if ("tag" in lnurlDetails) {
```

Comment:
maybe we should add some comments here. 
why do we do that?

Reply:
Using the `isError` function here now is well which makes this case clearer I hope